src/components/questions/RecordUpload.tsx

```diff
@@ -0,0 +1,158 @@
+import AudioRecorderPlayer, {
+  AVEncoderAudioQualityIOSType,
+  AVEncodingOption,
+  AudioEncoderAndroidType,
+  AudioSet,
+  AudioSourceAndroidType,
+  RecordBackType,
+  PlayBackType,
+} from 'react-native-audio-recorder-player';
+import { Alert, PermissionsAndroid, Platform, StyleSheet, View } from 'react-native';
+import React, { Component } from 'react';
+
+import Icon from 'react-native-vector-icons/FontAwesome';
+import COLORS from '../../assets/colors';
+
+interface State {
+  isPlaying: boolean;
+  isRecording: boolean;
+  onVoiceSelected: string;
+}
+class RecordUpload extends Component<any, State> {
+  private audioRecorderPlayer: AudioRecorderPlayer;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      isRecording: false,
+      isPlaying: false,
+      onVoiceSelected: '',
+    };
+
+    this.audioRecorderPlayer = new AudioRecorderPlayer();
+    this.audioRecorderPlayer.setSubscriptionDuration(0.1);
+  }
+
+  public render() {
+    return (
+      <>
+        <View style={styles.container}>
+          <View style={styles.rowContainer}>
+            {this.state.isRecording ? (
+              <Icon name="stop" size={48} style={styles.icon} onPress={this.onStopRecord} />
+            ) : (
+              <Icon name="microphone" size={48} style={styles.icon} onPress={this.onStartRecord} />
+            )}
+            {this.state.isPlaying ? (
+              <Icon name="pause" size={48} style={styles.icon} onPress={this.onPausePlay} />
+            ) : (
+              <Icon name="play" size={48} style={styles.icon} onPress={this.onStartPlay} />
+            )}
+            <Icon name="backward" size={48} style={styles.icon} onPress={this.onStopPlay} />
+            <Icon name="trash" size={48} style={styles.icon} onPress={this.onRemoveRecord} />
+          </View>
+        </View>
+      </>
+    );
+  }
+
+  private onStartRecord = async () => {
+    this.onStopPlay();
+    if (Platform.OS === 'android') {
+      try {
+        await PermissionsAndroid.requestMultiple([
+          PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+          PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
+          PermissionsAndroid.PERMISSIONS.RECORD_AUDIO,
+        ]);
+      } catch (err) {
+        console.warn(err);
+        return;
+      }
+    }
+    const audioSet: AudioSet = {
+      AudioEncoderAndroid: AudioEncoderAndroidType.AAC,
+      AudioSourceAndroid: AudioSourceAndroidType.MIC,
+      AVEncoderAudioQualityKeyIOS: AVEncoderAudioQualityIOSType.high,
+      AVNumberOfChannelsKeyIOS: 2,
+      AVFormatIDKeyIOS: AVEncodingOption.aac,
+    };
+    this.setState({
+      onVoiceSelected: await this.audioRecorderPlayer.startRecorder(undefined, audioSet),
+    });
+
+    this.audioRecorderPlayer.addRecordBackListener((e: RecordBackType) => {
+      console.log('record-back', e);
+      this.setState({
```

Comment:
Waarom wordt die elke callback gezet? En waarom wordt deze standaard op true gezet ipv dan de waarde uit de callback gebruiken

Reply:
Om te swichen tussen mic en stop button, voor in de view. Als het gaat om isRecording.

Reply:
Ja snap ik, maar waarom moet dit elke callback gebeuren? Je gaat er op dit moment vanuit dat als je op record drukt de recording daadwerkelijk bezig is en dus kan je deze buiten de callback ook gewoon op true zetten. Wil je het wel in de callback zelf doen, gebruik dan de waarde uit RecordBackType. Je gaat er nu vanuit dat de opname bezig is, terwijl dit helemaal niet zo hoeft te zijn. in RecordBackType kan isRecording ook gewoon op false staan